app/controllers/pageflow/application_controller.rb

```diff
@@ -63,5 +68,14 @@ def after_sign_out_path_for(resource_or_scope)
     def locale_from_accept_language_header
       http_accept_language.compatible_language_from(I18n.available_locales)
     end
+
+    private
+
+    def debug_log(exception)
+      Rails.logger.debug exception
+      exception.backtrace.each do |line|
+        Rails.logger.debug line
+      end
+    end
```

Comment:
Still a bit undecided if this is really an issue, but defining methods in `ApplicationController` - even if they are private - makes them available in all of Pageflow's controllers risking collisions if ever a controller should decide to implement a `debug_log` method itself.

This could be circumvented by turning `debug_log` into a module function of a nested module. Since modules are lexically scoped, there would be no problem when a controller defines a nested module by the same name.

Reply:
So maybe just given the method a more explicit name like "debug_log_exception" might be a compromise.

Also the comments to the answer of this SO question give some good arguments against calling `logger.debug` in a loop.

Reply:
Is there a link to SO missing, or did I just not find it?

Reply:
Indeed. Here it is: https://stackoverflow.com/questions/3484337/rails-logging-the-entire-stack-trace-of-an-exception